src/lib/rnp.c

```diff
@@ -1198,16 +888,17 @@ rnp_match_pubkeys(rnp_t *rnp, char *name, void *vp)
 
 /* find a key in a keyring */
 int
-rnp_find_key(rnp_t *rnp, char *id)
+rnp_find_key(rnp_t *rnp, const char *id)
 {
     pgp_io_t *io;
 
     io = rnp->io;
     if (id == NULL) {
         (void) fprintf(io->errs, "NULL id to search for\n");
-        return 0;
+        return RNP_FAIL;
     }
-    return rnp_key_store_get_key_by_name(rnp->io, rnp->pubring, id) != NULL;
+    return rnp_key_store_get_key_by_name(rnp->io, rnp->pubring, id) != NULL ? RNP_OK :
```

Comment:
It's again int vs bool. I will fix this method to return bool, since it is more suitable here.